### PR TITLE
Leetcode Question Daily Temperatures 

### PR DESCRIPTION
**Pull request by abethegr8**
Leetcode Question Daily Temperatures, working with Stacks, Monotonic Stacks. 
https://leetcode.com/problems/daily-temperatures/

### CODE REVIEW

File: src/DailyTemperature.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+Evaluate: Time O(N) and Space O(N)
+
+ */
+
+import java.util.Stack;
+
+public class DailyTemperature {
+    public static int[] dailyTemperature(int[] temperature){
+        int N = temperature.length;
+        int[] daysForTemperatureChange = new int[N];        // We will return this array. Will hold the amount of days we have to wait for it to be warmer
```
Comment (jchaffraix) [inline src/DailyTemperature.java:71]: The algorithm relies on array values being initialized to 0 ([Oracle documentation](https://docs.oracle.com/javase/specs/jls/se7/html/jls-4.html#jls-4.12.5)). I think this should be called out specifically (or else we would have to manually empty the stack after the for-loop on l. 84).

File: src/DailyTemperature.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        Stack<Integer> indexOfEachTemperatureArrayElement = new Stack<>();      // Monotonic Stack, keeps the index of the temperature of that day
```
Comment (jchaffraix) [inline src/DailyTemperature.java:72]: In general, you should avoid "what" comment (like "this keeps the index of the temperature of that day") and focus more on the "why".

Most "what" comments can usually be removed by using [self-documenting code](https://en.wikipedia.org/wiki/Self-documenting_code). In this case, we can rename the variable to `indexesOfHighTemperatureDays` (or `highTemperatureDaysStack` or something equivalent): this conveys what is written as a comment directly.

File: src/DailyTemperature.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        for (int i = 0; i < N; i++) {
+            // Go into the while loop if stack is not empty and the temperature tomorrow is warmer than today
```
Comment (jchaffraix) [inline src/DailyTemperature.java:75]: This comment adds no value as it is exactly mirroring what the code is doing on the next line. You should assume reasonable code readability from reviewers/readers.

By the way, this is a "what" comment.

File: src/DailyTemperature.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+else, lets just push the currentday into the stack. stack holds the current day which is the index of the temperature array
+stack.push(i);
+finally lets just return the array called answer
+
+Implement: see method below
+
+Review: review any bugs in the code, optimize if asked to.
+
+Evaluate: Time O(N) and Space O(N)
+
```
Comment (jchaffraix) [inline src/DailyTemperature.java:63]: Not covered by UMPIRE is testing. I really think we should list some tests that you considered.

File: src/DailyTemperature.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+ */
+
+import java.util.Stack;
+
+public class DailyTemperature {
+    public static int[] dailyTemperature(int[] temperature){
+        int N = temperature.length;
```
Comment (jchaffraix) [inline src/DailyTemperature.java:70]: `N` is OK in this exercise as there is only one size but beware that it is easy to be confused. You can map this to a more meaningful name like `temperatureSize` or `numberOfDays` (we should probably rename `temperate` to `temperatePerDay` to make it clear those are aligned though).

File: src/DailyTemperature.java (hunk 3 above)
Comment (abethegr8) [reply src/DailyTemperature.java:75]: Removed what comment.

File: src/DailyTemperature.java (hunk 1 above)
Comment (jchaffraix) [reply src/DailyTemperature.java:71]: I know and that wasn't the point.

The point is that this algorithm relies on those default values. In other language (e.g. C++ or Rust), memory is not zero'd out by default and as such such your algorithm would have undefined behavior. That's why it's important to call out that we rely on the default value.

**Review by jchaffraix — COMMENTED** [review]
Let's agree moving forward on not merging unreviewed code as I think we could have improved on the code again.

The goal of those reviews is to show you improvements so you can develop a sense for what great code looks like.

**Pull request merged**